### PR TITLE
oneMKL: support for scal

### PR DESCRIPTION
**Pull request by kballeda**
This PR contains support for scal primitive of oneMKL - level1

### CODE REVIEW

File: test/onemkl.jl [hunk 1]
```diff
@@ -27,19 +27,15 @@ k = 13
             gpuA = oneArray(A)
             if T === ComplexF32
                 alphaf32 = rand(Float32, 1)
-                B = A .* alphaf32[1]
                 oneMKL.scal!(m, alphaf32[1], gpuA)
-                @test Array(B) == Array(gpuA)
+                @test isapprox(A .* alphaf32[1], Array(gpuA))
```
Comment (maleadt) [inline test/onemkl.jl:31]: Use `@test ... ≈ ...`

**Comment by kaliudayintel** [comment]
@maleadt  closing this PR temporarily as it requires some changes. We have axpy / nrm2 prs that can go in first.
I will be reopening this PR once the other PRs are through.

**Comment by maleadt** [comment]
There is no need to temporarily close a PR. If you'll re-open it, just leave it open and mark it as draft.

File: lib/mkl/oneMKL.jl [hunk 2]
```diff
@@ -12,6 +12,7 @@ using GPUArrays
 
 include("libonemkl.jl")
 
+# Exclude Float16 for now, since many oneMKL functions - copy, scal, do not take Float16
 const onemklFloat = Union{Float64,Float32,Float16,ComplexF64,ComplexF32}
```
[the remark this answers is not in the document]
Comment (kaliudayintel) [reply lib/mkl/oneMKL.jl]: Thanks for the review. I have removed it.

**Comment by maleadt** [comment]
Same here, why does this need to `wait`?

**Comment by Sarbojit2019** [comment]
@kballeda, wait() is no longer needed as we have added force flush "__FORCE_MKL_FLUSH__(status)" on every submission. Please update the PR accordingly.

**Comment by kballeda** [comment]
Updated with force_mkl_flush instead of wait()

**Comment by maleadt** [comment]
LGTM, thanks.

**Pull request merged**